projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_configuration.py

```diff
@@ -26,7 +33,7 @@ def authentication_type(self):
 
     def keytab_folder(self):
         keytab_folder = self.__config.get_value(KEYTAB_FOLDER)
-        if keytab_folder is None:
+        if keytab_folder is None and self.__job_directory:
             keytab_folder = Path(self.__job_directory).parent
         return keytab_folder
```

Comment:
What happens if keytab_folder is None but also self.__job_directory doesn't exist? There might be reason why this isn't possible that I'm missing, but otherwise it seems the whole function will return None.

Reply:
I think I have the opposite question: Is it possible for `__job_directory` to not exist?

Reply:
Added tests to cover some of those scenarios.